lib/learn_to_rank/data_pipeline.rb

```diff
@@ -51,21 +54,27 @@ def self.can_run?
     end
 
     def self.get_status
-      redis = Redis.new(
-        host: ENV.fetch("REDIS_HOST", "127.0.0.1"),
-        port: ENV.fetch("REDIS_PORT", 6379),
-        namespace: "search-api-ltr",
-      )
-      redis.get("ltr-status") || READY_STATUS
+      redis.get(REDIS_STATUS_KEY) || READY_STATUS
     end
 
     def self.set_status(status)
-      redis = Redis.new(
+      redis.setex(REDIS_STATUS_KEY, REDIS_EXPIRATION_PERIOD, status)
+    end
+
+    def self.get_latest_data
+      redis.get(REDIS_LATEST_DATA_KEY)
+    end
+
+    def self.set_latest_data(data)
+      redis.set(REDIS_LATEST_DATA_KEY, data)
+    end
+
+    def self.redis
+      Redis.new(
         host: ENV.fetch("REDIS_HOST", "127.0.0.1"),
         port: ENV.fetch("REDIS_PORT", 6379),
-        namespace: "search-api-ltr",
+        namespace: REDIS_NAMESPACE,
```

Comment:
Do the search machines share a single redis instance, or do they have one instance per machine?

Reply:
I believe they share a redis instance